### PR TITLE
logger: output footer once for multithreaded push/pull errors

### PR DESCRIPTION
**Pull request by n3hrox**
* [x] Have you followed the guidelines in our
      [Contributing document](https://dvc.org/doc/user-guide/contributing/core)?

* [x] Does your PR affect documented changes or does it add new functionality
      that should be documented? If yes, have you created a PR for
      [dvc.org](https://github.com/iterative/dvc.org) documenting it or at
      least opened an issue for it? If so, please add a link to it.

-----

This should fix https://github.com/iterative/dvc/issues/2553
There is one test already for `_download` so it should be sufficient.
If you have any suggestions how could I test the footer appearance deduplication with pytest please let me know! This is my first PR for open source repo so any comments are welcome.

### CODE REVIEW

File: dvc/remote/base.py [hunk 1]
```diff
@@ -555,9 +553,7 @@ def download(
                 from_info, tmp_file, name=name, no_progress_bar=no_progress_bar
             )
         except Exception:
-            msg = "failed to download '{}' to '{}'"
-            logger.exception(msg.format(from_info, to_info))
-            return 1  # 1 fail
+            return 1, from_info, to_info, "download"  # 1 fail
```
[the remark this answers is not in the document]
Comment (n3hrox) [reply dvc/remote/base.py]: You are right @efiop, your solution is way better. Sorry I didn't notice the general exception handling inside `main.py`
I will refactor this over the next few days when I find some free time, hope you don't mind the wait

[the remark this answers is not in the document]
Comment (efiop) [reply dvc/remote/base.py]: Sure! Thank you so much for working on this!

File: dvc/main.py [hunk 2]
```diff
@@ -75,6 +86,9 @@ def main(argv=None):
         # so won't be reused by any other subsequent run anyway.
         clean_repos()
 
+    if ret != 0:
```
Comment (n3hrox) [inline dvc/main.py]: not sure if it should be here or rather in `finally` block

Comment (Suor) [reply dvc/main.py]: This place is fine.

File: dvc/main.py [hunk 3]
```diff
@@ -16,6 +17,16 @@
 
 logger = logging.getLogger("dvc")
 
+footer = (
```
Comment (Suor) [inline dvc/main.py]: I would create this message only when it's needed. Maybe some `show_troubles_footer()` func.

Comment (efiop) [reply dvc/main.py]: @Suor There is no point to do that. This message is meant to be static.

Comment (efiop) [inline dvc/main.py]: Also, let's call it
```suggestion
FOOTER = (
```
so it conforms to the naming conventions :slightly_smiling_face: `logger` from above does not, but that is a common practice for that specific application.

File: dvc/main.py [hunk 4]
```diff
@@ -75,6 +86,9 @@ def main(argv=None):
         # so won't be reused by any other subsequent run anyway.
         clean_repos()
 
+    if ret != 0:
+        logger.info(footer)
```
Comment (efiop) [inline dvc/main.py]: We used to print that message to stderr, and now we are going to print it to stdout. It would be great if we could keep it in stderr, if it requires a reasonable amount of effort. Otherwise, we could keep it in stdout as is for now.

Also, just wanted to point this out for the record, this helper message won't be shown to anyone using dvc through API now, and I think that is acceptable, as its primary purpose is for the CLI users. We might consider bringing these back in the future if we publically release such a low-level API.

File: dvc/main.py (hunk 3 above)
Comment (Suor) [reply dvc/main.py]: @efiop but it's created on each run for no purpose.

Comment (efiop) [reply dvc/main.py]: Not sure I follow, could you elaborate?

Comment (efiop) [reply dvc/main.py]: Ah, I guess you mean ColorFormatter instance:slightly_frowning_face: Good call.

Comment (efiop) [reply dvc/main.py]: Ah, wait, it is not. You meant that it is created for each `dvc` CLI command run. That is totally fine, don't see anything wrong with it.

Comment (Suor) [reply dvc/main.py]: I mean we import `colorama` and construct the message even when we'll never need it. I was thinking about startup time. However, since we import `colorama` in `dvc.logger` anyway then it probably doesn't matter for now.

File: dvc/logger.py [hunk 5]
```diff
@@ -66,18 +67,14 @@ def format(self, record):
...
-                "{stack_trace}\n"
-                "\n"
-                "{footer}"
+                "{color}{levelname}{nc}: {description}" "{stack_trace}\n"
             ).format(
                 color=self.color_code.get(record.levelname, ""),
                 nc=colorama.Fore.RESET,
                 levelname=record.levelname,
                 description=self._description(record.msg, exception),
                 msg=record.msg,
```
Comment (n3hrox) [inline dvc/logger.py]: I see that deepsource detected that `msg` is not used here although I did not delete it
I think adding it back to message or removing it entirely should be scope of another PR to be honest, as it affects multiple test cases. 
What do you think @efiop?

Comment (efiop) [reply dvc/logger.py]: You are right, it is out of scope for this PR.
